packages/react/src/xcss-prop/index.ts

```diff
@@ -27,19 +27,31 @@ type XCSSPseudos<
     : never;
 };
 
+type XCSSMediaQuery<
+  TAllowedProperties extends keyof CSSProperties,
+  TAllowedPseudos extends CSSPseudos,
+  TAllowedMediaQueries extends string,
+  TRequiredProperties extends { requiredProperties: TAllowedProperties },
+  TSchema
+> = {
+  [Q in `@media ${TAllowedMediaQueries}`]?:
+    | MarkAsRequired<
+        XCSSItem<TAllowedProperties, Q extends keyof TSchema ? TSchema[Q] : object>,
+        TRequiredProperties['requiredProperties']
+      >
+    | XCSSPseudos<TAllowedProperties, TAllowedPseudos, TRequiredProperties, TSchema>;
+};
+
 /**
  * These APIs we don't want to allow to be passed through the `xcss` prop but we also
  * must declare them so the (lack-of a) excess property check doesn't bite us and allow
  * unexpected values through.
  */
 type BlockedRules = {
   selectors?: never;
+  // We also block all type level at rule "objects" that are present on cssMap.
 } & {
-  /**
-   * We currently block all at rules from xcss prop.
-   * This needs us to decide on what the final API is across Compiled to be able to set.
-   */
-  [Q in CSS.AtRules]?: never;
+  [Q in AtRules]?: never;
```

Comment:
How does this work `XCSSMediaQuery & BlockedRules`? 🤔

Reply:
Last one wins.